test/packet/scripts/add_vagrant_box.sh

```diff
@@ -1,55 +1,169 @@
-    name=$(cat $path | grep "^\$${box}_BOX" | awk '{print $NF}' | sed 's/^"\(.*\)"$/\1/')
-    # we need non-dev images for CI
-    if [[ "$name" == "cilium/ubuntu-dev" ]]; then
-        name="cilium/ubuntu"
+    current_version=$(vagrant box list | awk '/'$box' /{sub(/)/,"",$3);if($3>v){v=$3}} END{if(v)print v;else print "0"}')
+
+    if (($current_version >= $latest_version)) ; then
+        echo -e "$box:\tlocal version $current_version >= remote version $latest_version, no need to preload"
+        version=0
+    else
+        echo -e "$box:\tlocal version $current_version, remote version $latest_version needs to be preloaded"
+        version=$latest_version
     fi
+}
 
-    if [[ "$name" == "" ]]; then
+for box in $boxes; do
+    if [[ latest -eq 1 ]] ; then
+        check_latest_version $box
+    else
+        check_defaults_version $box
+    fi
+    if [[ version -eq 0 ]] ; then
         continue
     fi
 
-    version=$(cat $path |grep "^\$${box}_VERSION" | awk '{print $NF}' | sed 's/^"\(.*\)"$/\1/')
-
     set +e
-	vagrant box list | grep "$name " | grep $version
-	if [[ $? -eq 0 ]]; then
-		echo "box already exists, no need to preload"
-		continue
-	fi
-
-    curl --fail http://vagrant-cache.ci.cilium.io/$name/$version/lock
+    curl --fail "$vagrant_url$box/$version/lock"
     lock_exit_code=$?
     # check for 404 error indicating that cache is up and lock file is not in place
-    if [ $lock_exit_code  -eq 22 ]; then
+    if [ $lock_exit_code -eq 22 ]; then
         download_from_cache=true
     else
         # cache is down or box is locked
         download_from_cache=false
     fi
 
-    box_downloaded=false
+    ret=1
 
     if [[ $download_from_cache == true ]]; then
         echo "adding box from cache"
-        curl --fail http://vagrant-cache.ci.cilium.io/$name/$version/metadata.json --output metadata.json
-        curl http://vagrant-cache.ci.cilium.io/$name/$version/package.box --output package.box
-        vagrant box add metadata.json
-        box_downloaded=$?
+        curl --fail "$vagrant_url$box/$version/metadata.json" -o $outdir/metadata.json
+        url="$vagrant_url$box/$version/package.box"
+        if [[ $use_aria2 -eq 1 ]] ; then
+            aria2c -x16 -s16 -c -d "$outdir" -o package.box "$url"
+            ret=$?
+        else
+            curl "$url" -o "$outdir/package.box"
+            ret=$?
+        fi
+        if [[ $ret -eq 0 ]]; then
+            cd $outdir
+            vagrant box add metadata.json
+            ret=$?
+        fi
     fi
 
     set -e
-    if [[ $box_downloaded -ne 0 ]]; then
+    if [[ $ret -ne 0 ]]; then
         echo "box locked or unavailable, adding box from vagrant cloud"
-        vagrant box add $name --box-version $version
+        if [[ $use_aria2 -eq 1 ]] ; then
+            aria2c -x16 -s16 -c -d "$outdir" -o package.box "$url"
+            vagrant box add "cilium/$box" "$outdir/package.box"
+            mv $box_dir$box/{0,$version}
```

Comment:
The `metadata_url` file also needs to be created here, for the first VM downloaded. Otherwise, Vagrant complains when trying to start the VM.

Reply:
Ah right, thanks. I'll add it.

Reply:
Just to be clear, it needs to be created in the `$box_dir$box/`